va/drm/va_drm_auth_x11.c

```diff
@@ -25,9 +25,16 @@
 #define _GNU_SOURCE 1
 #include "sysdeps.h"
 #include <dlfcn.h>
-#include <X11/Xlib.h>
+#include <stdint.h>
 #include "va_drm_auth_x11.h"
 
+/* A few ABI compabible defines and typedefs to avoid pulling Xlib.h */
```

Comment:
If we need Xlib.h, then we need to include it rather than mimic it. If we don't need to link against the library, then don't link, but add paths to include files - this should be managed on build system level.

Reply:
I would greatly encourage you to take a dive in the the Xlib.h headers. They are one of the worst one that exist. In practice this is the least crazy thing to do here.

Reply:
Sounds like throwing adjectives, instead of describing the details was a little misleading. Let me try again:

Various parts of X, predate standard C89/C99 and as such they often had to invent them to a varying degree. In some cases the X implementations would ignore/violate C standards be compatible on the API level but vary in ABI. Furthermore often the whole internal X ABI would be leaked and thus embedded into unsuspecting programs.

In this case here, we need a few very well defined and stable (from CPU arch or *nix flavour POV) types. We simply do not need the chaos that `#include <X11/Xlib.h>` brings.

For the same reasons, we've done similar changes in Mesa, so I believe we could and should really do the same in here. Does that make sense?

Reply:
@dvrogozh ping?

Reply:
@XinfengZhang @dvrogozh humble poke